subgraphs/uniswap-forks/src/common/getters.ts

```diff
@@ -55,8 +56,44 @@ export function getOrCreateProtocol(): DexAmmProtocol {
   return protocol;
 }
 
-export function getLiquidityPool(poolAddress: string): LiquidityPool {
-  return LiquidityPool.load(poolAddress)!;
+export function getLiquidityPool(
+  poolAddress: string,
+  blockNumber: BigInt
+): LiquidityPool {
+  const pool = LiquidityPool.load(poolAddress)!;
+  pool.fees = updateFeesIfChanged(poolAddress, blockNumber);
+  pool.save();
+  return pool;
+}
+
+export function updateFeesIfChanged(
```

Comment:
Do you think the ‘createPoolFees’ and ‘updatePoolFeesIfChanged’ functions could be made one? Like a ‘getOrCreatePoolFees’. Also, I think the ‘LiquidityPoolFee’ entities should be loaded in and not created new in ‘updatePoolFeesIfChanged’. What do you think?

Reply:
Agreed, I recommended the same in discord discussion but forgot doing that here, will fix :)